src/RemoteInboxNotifications/ComparisonOperation.php

```diff
@@ -19,6 +19,18 @@ class ComparisonOperation {
 	 * @param string $operation     The operation used to compare the operands.
 	 */
 	public static function compare( $left_operand, $right_operand, $operation ) {
+		if ( $operation && strpos( $operation, 'contains' ) !== false && ! is_array( $left_operand ) ) {
+			$logger = wc_get_logger();
```

Comment:
**Optional**

I think it's good how this won't fail silently, but I feel like the log message doesn't provide enough info to help someone troubleshoot the issue.

You could add logging earlier when the `woocommerce_task_list_tracked_completed_tasks` option is first retrieved, and check if it is an array then. This would allow a more specific message to be written about the issue.

Reply:
Yeah that makes sense, will move this back.

Reply:
I moved it to the `OptionsRuleProcessor` instead adding the option name in the warning: https://github.com/woocommerce/woocommerce-admin/pull/6645/commits/a4956639f65384a267be0befeb33fb4225a28fda